### ISSUE TITLE
If a data only container starts with the same name than the active container, compose fails

### ISSUE THREAD

**Issue by MBuffenoir** [docker/compose#1454]
Example docker-compose.yml:

    dbdata:
      image: postgres:9.3
      command: echo "data only container (might already exists)"

    db:
      image: postgres:9.3
      environment:
        DB_PASSWORD: postgres
        DB_USER: postgres
        DB_NAME: postgres
      ports:
        - "5432:5432"
      volumes_from: 
        - dbdata

Error raised is:
A service can not mount itself as volume

I believe the error is due to the usage of "in" here:
https://github.com/docker/compose/blob/master/compose/project.py#L38

**Comment by thaJeztah** [comment]
Ah, interesting :)

@aanand I wonder if this can be tweaked as well, now that labels are used?

**Comment by kojiromike** [comment]
I couldn't reproduce this behavior with docker-compose 1.3.0 dev (889d363). I tried this simplified test case:

```yaml
aa:
    image: busybox
    volumes:
        - /foo
    command: echo 'oof'
a:
    image: busybox
    volumes_from:
        - aa
```

And then I tried OP's original example. I tried with both `up` and `run`. I did not get an error in any of those four attempts.

**Comment by dnephin** [comment]
The "in" on that line means "in the list" not "not in the string".  

If this is still an issue please feel free to re-open with an update config and the full output of the `docker-compose` command you ran.

**Comment by richfletcher** [comment]
For the benefit of future searchers, this appears to be caused when a string is used as the value of volumes_from instead of a list.  Then, the 'in' above checks for a substring of the given value for volumes_from.

Of course, a possible enhancement could be to either refuse a string or automatically listify it.

**Comment by dnephin** [comment]
Ah, this should be fixed by #129
